dot/network/warp_sync.go

```diff
@@ -55,7 +58,20 @@ func (s *Service) handleWarpSyncMessage(stream libp2pnetwork.Stream, msg message
 		}
 	}()
 
+	reqId := fmt.Sprintf("%s-%s", stream.Conn().RemotePeer(), msg.String())
+	hashedreqId := common.MustBlake2bHash([]byte(reqId))
+
 	if req, ok := msg.(*messages.WarpProofRequest); ok {
+		// Check if this peer has exceeded the limit of requests
+		if !s.warpSyncSpamLimiter.IsLimitExceeded(hashedreqId) {
```

Comment:
Why is this negated? Shouldn't this be `if s.warpSyncSpamLimiter.IsLimitExceeded(hashedreqId) {`?

Reply:
Yes!! thanks for the catch!